src/java.base/share/classes/jdk/internal/reflect/MethodHandleFieldAccessorImpl.java

```diff
@@ -68,17 +68,15 @@ protected final void ensureObj(Object o) {
      * IllegalArgumentException because Field::get on the specified object, which
      * is not an instance of the class or interface declaring the underlying method
      */
-    protected IllegalArgumentException newGetIllegalArgumentException(Class<?> type) {
-        return new IllegalArgumentException(getMessage(true, type.getName()));
+    protected IllegalArgumentException newGetIllegalArgumentException(Object o) {
+        return new IllegalArgumentException(getMessage(true, o != null ? o.getClass().getName() : ""));
```

Comment:
Hello Mandy,
With this change the `getMessage` may get passed an empty string, so it would end up printing something like `Can not get ... field on`. Do you think the `getMessage` implementation should be tweaked not to print the "on" if the `attemptedType` is empty?

Reply:
> Hello Mandy,
> With this change the `getMessage` may get passed an empty string, so it would end up printing something like `Can not get ... field <fieldname> on`. Do you think the `getMessage` implementation should be tweaked not to print the "on" if the `attemptedType` is empty?

I suspect this was meant to be "null" rather than "".

Reply:
With this patch, `newGetIllegalArgumentException` and `newSetIllegalArgumentException` are now only called when obj is non-null and the receiver type is of an invalid type.    But it's better to improve this exception message to handle null obj.    I pushed the change to clean up this code.